### PR TITLE
`data` made optional in public API

### PR DESCRIPTION
**Pull request by HarsheetKakar**
#### References to other Issues or PRs or Relevant literature

solves issue referenced in [this](https://github.com/codezonediitj/pydatastructs/pull/243#discussion_r403913053) comment

### CODE REVIEW

File: pydatastructs/trees/heaps.py [hunk 1]
```diff
@@ -146,6 +146,9 @@ def insert(self, key, data):
 
         None
         """
+        if data is None:
+            data = key
+
```
Comment (czgdp1807) [inline pydatastructs/trees/heaps.py:151]:
```suggestion
```

File: pydatastructs/trees/tests/test_heaps.py [hunk 2]
```diff
@@ -11,7 +11,7 @@ def test_BinaryHeap():
 
     assert raises(IndexError, lambda: max_heap.extract())
 
-    max_heap.insert(100, 100)
+    max_heap.insert(100)
```
Comment (czgdp1807) [inline pydatastructs/trees/tests/test_heaps.py:14]:
```suggestion
    max_heap.insert(100, 100)

```

**Comment by HarsheetKakar** [comment]
merge it so that I can make changes in heapsort as well

**Comment by czgdp1807** [comment]
Apply [this diff file](https://github.com/codezonediitj/pydatastructs/files/4439120/pds_diff.txt).

**Pull request merged**